learn2learn/algorithms/maml.py

```diff
@@ -165,6 +169,23 @@ def adapt(self,
                 traceback.print_exc()
                 print('learn2learn: Maybe try with allow_nograd=True and/or allow_unused=True ?')
 
+        # Clip the gradients in place if the argument is set, using the default L2 norm
+        # (code mainly copied from
+        # https://pytorch.org/docs/stable/_modules/torch/nn/utils/clip_grad.html with some
+        # adjustments)
+        if clip_grad_max_norm:
+            norm_type = 2.0
+            device = gradients[0].device
+            total_norm = torch.norm(torch.stack(
+                [torch.norm(g.detach(), norm_type) for g in gradients if g is not None]
+            ).to(device), norm_type)
+            if total_norm.isnan() or total_norm.isinf():
+                warnings.warn("Non-finite norm encountered in learn2learn.algorithms.MAML.adapt; continuing anyway.", FutureWarning, stacklevel=2)
+            clip_coef = clip_grad_max_norm / (total_norm + 1e-6)
+            if clip_coef < 1:
+                for g in gradients:
+                    if g is not None:
+                        g.detach().mul_(clip_coef.to(g.device))
```

Comment:
Do we need detach here? It would break the computation graph of the meta-gradients.